crates/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -759,23 +765,37 @@ impl ManagedEnvironment {
                 .floxmeta
                 .git
                 .branch_contains_commit(
+                    &branch_name(&self.system, &self.pointer, &self.path).unwrap(),
                     &remote_branch_name(&self.system, &self.pointer),
-                    "FETCH_HEAD",
                 )
                 .map_err(ManagedEnvironmentError::Git)?;
             if !consistent_history {
                 Err(ManagedEnvironmentError::Diverged)?;
             }
         }
 
+        // try fast forward merge local env branch into project branch
+        self.floxmeta
+            .git
+            .push_ref(
+                ".",
+                format!(
+                    "refs/heads/{sync_branch}:refs/heads/{project_branch}",
+                    sync_branch = remote_branch_name(&self.system, &self.pointer),
```

Comment:
nonblocking: could add `let sync_branch =` at the top of the function

Reply:
yeah there are too many now